### PR TITLE
Security

### PR DESCRIPTION
**Pull request by alexandralopez98**

### CODE REVIEW

File: src/main/java/codeu/controller/RegisterServlet.java [hunk 1]
```diff
@@ -46,7 +47,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
-    User user = new User(UUID.randomUUID(),username,password,Instant.now());
+    User user = new User(UUID.randomUUID(),username,passwordHash,Instant.now());
```
Comment (ElleTojaroon) [inline src/main/java/codeu/controller/RegisterServlet.java]: nit: leave a space after every comma

**Comment by ElleTojaroon** [comment]
Will there be another commit for affected tests?

**Comment by alexandralopez98** [comment]
Yes, once I figure them out I will make another commit!

**Comment by ElleTojaroon** [comment]
@alexandralopez98  I fixed the test errors. Now all tests pass. See the changes. Main things I spotted:
1. User constructor encrypt the password, which is already encrypted in its caller such as RegisterServlet.java. The stored password was then double encrypted. 
2. Instead of doing assertEquals() to compare the stored password with the expected one, we validate with assertTrue(BCrypt.checkpw(<expected password>, <hash password aka from the ArgumentCaptor>)
3. See RegisterServlet.java:37. Instead of encrypting the given password, the string "password" was encrypted. I changed that to request.getParameter("password"). 
4. I didn't check this thoroughly enough in past CR, but it's a good practice to have constant fields instead of literal strings. See my changes in LoginServletTest.java.

**Comment by alexandralopez98** [comment]
Thank you so much!

**Pull request merged**